casa_formats_io/casa_low_level_io.py

```diff
@@ -113,329 +216,889 @@ def read_type(f):
+            elif rectype == 'string':
+                self.values[name] = read_string(f)
+            elif rectype == 'table':
+                self.values[name] = 'Table: ' + os.path.abspath(os.path.join(f.original_filename, read_string(f)))
+            elif rectype == 'arrayint':
+                self.values[name] = read_array(f, 'int')
+            elif rectype == 'arrayuint':
+                self.values[name] = read_array(f, 'uint')
+            elif rectype == 'arrayfloat':
+                self.values[name] = read_array(f, 'float')
+            elif rectype == 'arraydouble':
+                self.values[name] = read_array(f, 'double')
+            elif rectype == 'arraycomplex':
+                self.values[name] = read_array(f, 'complex')
+            elif rectype == 'arraydcomplex':
+                self.values[name] = read_array(f, 'dcomplex')
+            elif rectype == 'arraystr':
+                self.values[name] = read_array(f, 'string')
+            elif rectype == 'record':
+                self.values[name] = TableRecord.read(f).values
+            else:
+                raise NotImplementedError("Support for type {0} in TableRecord not implemented".format(rectype))
+
+        return self
+
+    def as_dict(self):
+        return self.values
+
+
+class Table(BaseCasaObject):
+
+    @classmethod
+    def read(cls, filename, endian='<'):
+        """
+        Read a CASA table - note that this is currently **experimental** and
+        should not be used for production code.
+        """
+
+        with open(os.path.join(filename, 'table.dat'), 'rb') as f_orig:
+
+            f = EndianAwareFileHandle(f_orig, '>', filename)
+
+            magic = f.read(4)
+            if magic != b'\xbe\xbe\xbe\xbe':
+                raise ValueError('Incorrect magic code: {0}'.format(magic))
+
+            table = cls.read_fileobj(f)
+
+        for dm_index, dm in table.column_set.data_managers.items():
+
+            fx_filename = os.path.join(filename, f'table.f{dm_index}')
+
+            if os.path.exists(fx_filename):
+
+                with open(fx_filename, 'rb') as f_orig:
+
+                    if isinstance(dm, TiledShapeStMan):
+                        endian = '>'
+
+                    f = EndianAwareFileHandle(f_orig, endian, filename)
+
+                    magic = f.read(4)
+                    if magic != b'\xbe\xbe\xbe\xbe':
+                        raise ValueError('Incorrect magic code: {0}'.format(magic))
+
+                    dm.read_header(f)
+
+        table._filename = filename
+
+        return table
+
+    def read_as_astropy_table(self):
+
+        # We now loop over columns and read the relevant data from each bucket.
+
+        # TODO: refactor this to make it so that columns are read on request
+        # instead of all in one go. The best way to do this might be to make a
+        # table where each column is a dask array.
+
+        coldesc = self.desc.column_description
+
+        t = AstropyTable()
```

Comment:
minor: could you use a longer variable name, e.g., `tbl`?  (to make searching/highlighting easier)

Reply:
Done